eva_cttv_pipeline/clinvar_to_evidence_strings.py

```diff
@@ -1,280 +1,314 @@
-    n_multiple_allele_origin = 0
-    n_germline_somatic = 0
-    n_records_no_recognised_allele_origin = 0
-    no_variant_to_ensg_mapping = 0
-    n_more_than_one_efo_term = 0
-    n_same_ref_alt = 0
-    n_missed_strings_unmapped_traits = 0
-    n_nsvs = 0
-    n_valid_rs_and_nsv = 0
-    n_nsv_skipped_clin_sig = 0
-    n_nsv_skipped_wrong_ref_alt = 0
-    unmapped_traits = {}
-    n_unrecognised_allele_origin = {}
-    ensembl_gene_id_uris = set()
-    traits = set()
-    unrecognised_clin_sigs = set()
-    evidence_list = []  # To store Hellen Parkinson records of the form [RCV, rs, ClinVar trait, EFO url]
-    record_counter = 0
-    n_total_clinvar_records = 0
-    skip = 0
+    allowed_clinical_significance = allowed_clinical_significance.split(',') if allowed_clinical_significance else \
+        get_default_allowed_clincal_significance()
 
-    answer = urllib.request.urlopen('http://' + config.HOST + '/cellbase/webservices/rest/v3/hsapiens/feature/clinical/all?source=clinvar&skip=' + str(skip) + '&limit=' + str(config.BATCH_SIZE))
+    mappings = get_mappings(efo_mapping_file, ignore_terms_file, adapt_terms_file, snp_2_gene_file,
+                            variant_summary_file)
+
+    report = Report(unavailable_efo_dict=mappings.unavailable_efo_dict)
+
+    for cellbase_record in get_records(report.counters):
+
+        record = get_record(cellbase_record, mappings)
+        report.counters["record_counter"] += 1
+        report.counters["n_nsvs"] += (record.clinvarRecord.get_nsv(mappings.rcv_to_nsv) is not None)
+        append_nsv(report.nsv_list, record.clinvarRecord, mappings.rcv_to_nsv)
+
+        if skip_record(cellbase_record, record, allowed_clinical_significance, mappings.rcv_to_nsv, report.counters):
+            continue
+
+        ensembl_gene_ids = record.con_type.ensembl_gene_ids
+        traits = get_traits(record.clinvarRecord.traits, mappings.trait_2_efo, report)
+
+        report.counters["n_multiple_allele_origin"] += (len(record.clinvarRecord.allele_origins) > 1)
+        report.counters["n_germline_somatic"] += (('germline' in record.clinvarRecord.allele_origins) and (
+        'somatic' in record.clinvarRecord.allele_origins))
+        report.counters["n_records_no_recognised_allele_origin"] += (
+            ('germline' not in record.clinvarRecord.allele_origins) and
+            ('somatic' not in record.clinvarRecord.allele_origins))
+
+        for ensembl_gene_id, trait, allele_origin in itertools.product(ensembl_gene_ids, traits,
+                                                                       record.clinvarRecord.allele_origins):
+
+            ensembl_gene_id_uri = 'http://identifiers.org/ensembl/' + ensembl_gene_id
+            rcv_to_gene_evidence_codes = ['http://identifiers.org/eco/cttv_mapping_pipeline']  # Evidence codes provided by Mick
+
+            if allele_origin not in ('germline', 'somatic'):
+                report.n_unrecognised_allele_origin[allele_origin] += 1
```

Comment:
A non-critical matter of style, but maybe `continue` could be used to reduce the indentation of the `else` block?